.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,18 @@
+---
+name: pre-commit
+on:
+  pull_request:
+    branches:
+      - "s3gw"
```

Comment:
Is it deliberate not to check release branches? This might become important if we ever have to backport something and the change to the release branche ends up different than the change to the `s3gw` branch.

Reply:
I think it would be to avoid the annoyance of failing on the existing code base because it hasn't been fixed up until this point? Unless we want to backport the fix to all release branches (which I don't think we want to).

Reply:
The Pipeline isn't supposed to check the existing code base. Just the change set. Therefore it wouldn't fail on the existing release branches either unless you commit something in violation of the checkers.

Reply:
ah, okay. Then I'm with @m-ildefons on this, and we should be checking the release branches as well. :)

Reply:
ack